server/src/main/java/keywhiz/service/daos/GroupDAO.java

```diff
@@ -93,22 +120,70 @@ public void deleteGroup(Group group) {
           .set(SECRETS.OWNER, (Long) null)
           .where(SECRETS.OWNER.eq(group.getId()))
           .execute();
+      DSL.using(configuration)
+          .update(GROUPS)
+          .set(GROUPS.OWNER, (Long) null)
+          .where(GROUPS.OWNER.eq(group.getId()))
+          .execute();
     });
   }
 
   public Optional<Group> getGroup(String name) {
-    GroupsRecord r = dslContext.fetchOne(GROUPS, GROUPS.NAME.eq(name));
-    return Optional.ofNullable(r).map(groupMapper::map);
+    Record record = dslContext
+        .select(GROUPS.fields())
+        .select(GROUP_OWNERS.NAME)
+        .from(GROUPS)
+        .leftJoin(GROUP_OWNERS)
+        .on(GROUPS.OWNER.eq(GROUP_OWNERS.ID))
+        .where(GROUPS.NAME.eq(name))
+        .fetchOne();
+
+    return Optional.ofNullable(recordToGroup(record));
   }
```

Comment:
We can't rely on the convenience of a dumb mapper against a single table select, but I also don't think the new query is that much more complicated.